cala-ledger/src/transaction/mod.rs

```diff
@@ -0,0 +1,84 @@
+pub mod error;
+
+mod entity;
+mod repo;
+
+use sqlx::PgPool;
+use tracing::instrument;
+
+#[cfg(feature = "import")]
+use crate::primitives::DataSourceId;
+use crate::{entity::*, outbox::*, primitives::DataSource};
+
+pub use entity::*;
+use error::*;
+use repo::*;
+
+#[derive(Clone)]
+pub struct Transactions {
+    repo: TransactionRepo,
+    outbox: Outbox,
+    pool: PgPool,
+}
+
+impl Transactions {
+    pub(crate) fn new(pool: &PgPool, outbox: Outbox) -> Self {
+        Self {
+            repo: TransactionRepo::new(pool),
+            outbox,
+            pool: pool.clone(),
+        }
+    }
+
+    #[instrument(name = "cala_ledger.transactions.create", skip(self))]
+    pub async fn create(
```

Comment:
where is this being used? This should be illegal. You cannot create a transaction other than via `post_transaction` business logic... is this present in sqlx-ledger?

Reply:
IF this is needed to be called from `post_transaction` then it must be `pub(crate)` simple `pub` is too broad...